modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -779,21 +780,22 @@ static List<Index> getTargetIndices(
      */
     private void clearErrorStoreForUnmanagedIndices(DataStream dataStream) {
         Metadata metadata = clusterService.state().metadata();
-        for (String indexName : errorStore.getAllIndices()) {
+        final var projectId = metadata.getProject().id();
+        for (String indexName : errorStore.getAllIndices(projectId)) {
             IndexAbstraction indexAbstraction = metadata.getProject().getIndicesLookup().get(indexName);
```

Comment:
Nit:
```suggestion
            IndexAbstraction indexAbstraction = metadata.getProject(projectId).getIndicesLookup().get(indexName);
```
to keep deprecation in one place.

Reply:
I already have local changes ready to make `DataStreamLifecycleService` project aware - overriding these changes. So I'll leave this as is for now.